src/models/jwtPermissions/index.js

```diff
@@ -136,3 +155,48 @@ exports.init = async function ({ publicKeyUrl = _publicKeyUrl, loadPublicKeyFrom
         }
     }, 60 * 60 * 1000);
 };
+
+/**
+ *
+ * @param {string}tokenMerchantId
+ * @param {string}uriMerchantId
+ * @returns {void}
+ */
+const validateMerchant = (tokenMerchantId, uriMerchantId) => {
+    if (tokenMerchantId !== uriMerchantId && tokenMerchantId !== '*') {
+        throw new UserNotAuthorised(uriMerchantId);
+    }
+};
+
+/**
+ *
+ * @param ctx
+ * @param next
+ * @returns {Promise<void>}
+ */
+exports.validateMerchantMidleware = async (ctx, next) => {
+    const { jwtPayload } = ctx.state.jwtPayload;
+    const permisisonScope = getScope(jwtPayload, PERMISSION_SCOPE);
+    const tokenMerchantId = permisisonScope[2].merchantId;
+    const uriMerchantId = ctx.params.merchantId;
+    validateMerchant(tokenMerchantId, uriMerchantId);
+    await next();
+};
+
+/**
+ *
+ * @param {string}token
+ * @param {string}merchant
+ * @param {number[]|number}permissions
+ * @param { {string|undefined} }
+ * @returns {Promise<void>}
+ */
+exports.authorizeUser = async function (token, merchant, permissions, { publicKeyUrl = _publicKeyUrl } = {}) {
+    const payload = await validateJwt(token, publicKeyUrl);
+    const permisisonScope = getScope(payload, PERMISSION_SCOPE);
```

Comment:
typo - try to set your IDE to let you know about the typos, see the green line under the wrong word:
![image](https://user-images.githubusercontent.com/681900/132696023-a9d4ff1e-82dd-42c7-8d58-33d0dfe55404.png)

Reply:
done